client/client.go

```diff
@@ -1173,20 +1173,12 @@ func (c *Client) addAlloc(alloc *structs.Allocation) error {
 
 // setupConsulClient creates a ConsulService
 func (c *Client) setupConsulClient() error {
-	cfg := consul.ConsulConfig{
-		Addr:      c.config.ReadDefault("consul.address", "127.0.0.1:8500"),
-		Token:     c.config.Read("consul.token"),
-		Auth:      c.config.Read("consul.auth"),
-		EnableSSL: c.config.ReadBoolDefault("consul.ssl", false),
-		VerifySSL: c.config.ReadBoolDefault("consul.verifyssl", true),
-	}
-
-	cs, err := consul.NewConsulService(&cfg, c.logger, "")
+	cs, err := consul.NewConsulService(c.config.ConsulConfig, c.logger)
 	c.consulService = cs
 	return err
 }
 
-// syncConsul removes services of tasks which are no longer in running state
+// syncConsul removes services of tasks which are no longer in running state and
```

Comment:
I'm glad I'm not the only one who apparently does this every now and then and for reasons I can't explain. :~]